pylops/signalprocessing/dtcwt.py

```diff
@@ -0,0 +1,145 @@
+__all__ = ["DTCWT"]
+
+from typing import Union
+
+import dtcwt
+import numpy as np
+
+from pylops import LinearOperator
+from pylops.utils._internal import _value_or_sized_to_tuple
+from pylops.utils.decorators import reshaped
+from pylops.utils.typing import DTypeLike, InputDimsLike, NDArray
+
+
+class DTCWT(LinearOperator):
+    r"""
+    Perform Dual-Tree Complex Wavelet Transform on a given array.
+
+    This operator wraps around :py:func:`dtcwt` package.
+
+    Parameters
+    ----------
+    dims: :obj:`int` or :obj:`tuple`
+        Number of samples for each dimension.
+    transform: :obj:`int`, optional
+        Type of transform 1D, 2D or 3D. Default is 1.
+    birot: :obj:`str`, optional
+        Level 1 wavelets to use. See :py:func:`dtcwt.coeffs.birot()`. Default is `"near_sym_a"`.
+    qshift: :obj:`str`, optional
+        Level >= 2 wavelets to use. See :py:func:`dtcwt.coeffs.qshift()`. Default is `"qshift_a"`
+    nlevels: :obj:`int`, optional
+        Number of levels of wavelet decomposition. Default is 3.
+    include_scale: :obj:`bool`, optional
+        Include scales in pyramid. See :py:func:`dtcwt.Pyramid`. Default is False.
+    dtype : :obj:`DTypeLike`, optional
+        Type of elements in input array.
+    name : :obj:`str`, optional
+        Name of operator (to be used by :func:`pylops.utils.describe.describe`)
+
+    Raises
+    ------
+    NotImplementedError
+        If ``transform`` is 2 or 3.
+    ValueError
+        If ``transform`` is anything other than 1, 2 or 3.
+
+    Notes
+    -----
+    The :py:func:`dtcwt` library uses a Pyramid object to represent the transform domain signal.
+    It has
+        `lowpass` (coarsest scale lowpass signal)
+        `highpasses` (complex subband coefficients for corresponding scales)
+        `scales` (lowpass signal for corresponding scales finest to coarsest)
+
+    To make the dtcwt forward() and inverse() functions compatible with pylops, the Pyramid object is
+    flattened out and all coeffs(highpasses and low pass signal) are appened into one array using the
```

Comment:
coeffs(highpasses and low pass signal) -> coefficents (high-pass and low pass coefficients)

Reply:
updated [f887031](https://github.com/PyLops/pylops/pull/495/commits/f887031ff0aed0edac70bed9bf1388e2dbfb4b70)